### PR TITLE
Switch default source version to 3.6

### PR DESCRIPTION
**Pull request by hamzaremmal**

### CODE REVIEW

File: tests/neg/i7294.check
Comment (hamzaremmal) [inline tests/neg/i7294.check:1]: I don't think that this is the correct error to show. It should keep the following message:
```scala
-- [E007] Type Mismatch Error: tests/neg/i7294.scala:7:18 --------------------------------------------------------------
7 |  case x: T => x.g(10) // error // error
  |               ^^^^^^^
  |               Found:    Any
  |               Required: T
  |
  |               where:    T is a type in given instance f with bounds <: foo.Foo
```

Comment (hamzaremmal) [reply tests/neg/i7294.check:1]: Related issue: #20992

**Comment by bishabosha** [comment]
you can't add a 3.7 mode before 3.7?

**Comment by hamzaremmal** [comment]
> you can't add a 3.7 mode before 3.7?

No, I checked with @odersky this afternoon and this is the way it is supposed to be done

**Comment by bishabosha** [comment]
> No, I checked with @odersky this afternoon and this is the way it is supposed to be done

it seems since 3.4 this was the case (it had 3.5 source versions), but until then this never happened, so I did find it strange

**Review by WojciechMazur — APPROVED** [review]

**Comment by Gedochao** [comment]
This has been decided to be included in the 3.6.0 release.

**Pull request merged**